From 47c61bc22285224ee24a7cbfa822c481f1155a81 Mon Sep 17 00:00:00 2001
From: Stefan Hajnoczi <stefanha@redhat.com>
Date: Tue, 7 Oct 2014 14:07:00 +0200
Subject: [PATCH 42/43] trace: add systemtap-initscript README file to RPM

Message-id: <1412690820-31016-12-git-send-email-stefanha@redhat.com>
Patchwork-id: 61616
O-Subject: [RHEL7.1 qemu-kvm PATCH 11/11] trace: add systemtap-initscript README file to RPM
Bugzilla: 1088112
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

The downstream systemtap-initscript integration is undocumented.  Fix
that with a brief guide containing usage instructions.

Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit d73b49dbbce7535531da5aa1dc640208f18979d2)
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

Conflicts:
	redhat/qemu-kvm.spec.template

This patch is forward-ported from RHEL 6 qemu-kvm.  RHEL 7's
qemu-kvm.spec.template is significantly different from RHEL 6 so
adjustments are necessary to include README.systemtap in the RPM.
---
 README.systemtap              | 43 +++++++++++++++++++++++++++++++++++++++++++
 redhat/qemu-kvm.spec.template |  3 ++-
 2 files changed, 45 insertions(+), 1 deletion(-)
 create mode 100644 README.systemtap

diff --git a/README.systemtap b/README.systemtap
new file mode 100644
index 0000000..ad913fc
--- /dev/null
+++ b/README.systemtap
@@ -0,0 +1,43 @@
+QEMU tracing using systemtap-initscript
+---------------------------------------
+
+You can capture QEMU trace data all the time using systemtap-initscript.  This
+uses SystemTap's flight recorder mode to trace all running guests to a
+fixed-size buffer on the host.  Old trace entries are overwritten by new
+entries when the buffer size wraps.
+
+1. Install the systemtap-initscript package:
+  # yum install systemtap-initscript
+
+2. Install the systemtap scripts and the conf file:
+  # cp /usr/share/qemu-kvm/systemtap/script.d/qemu_kvm.stp /etc/systemtap/script.d/
+  # cp /usr/share/qemu-kvm/systemtap/conf.d/qemu_kvm.conf /etc/systemtap/conf.d/
+
+The set of trace events to enable is given in qemu_kvm.stp.  This SystemTap
+script can be customized to add or remove trace events provided in
+/usr/share/systemtap/tapset/qemu-kvm-simpletrace.stp.
+
+SystemTap customizations can be made to qemu_kvm.conf to control the flight
+recorder buffer size and whether to store traces in memory only or disk too.
+See stap(1) for option documentation.
+
+3. Start the systemtap service.
+　# service systemtap start qemu_kvm
+
+4. Make the service start at boot time.
+　# chkconfig systemtap on
+
+5. Confirm that the service works.
+  # service systemtap status qemu_kvm
+  qemu_kvm is running...
+
+When you want to inspect the trace buffer, perform the following steps:
+
+1. Dump the trace buffer.
+  # staprun -A qemu_kvm >/tmp/trace.log
+
+2. Start the systemtap service because the preceding step stops the service.
+  # service systemtap start qemu_kvm
+
+3. Translate the trace record to readable format.
+  # /usr/share/qemu-kvm/simpletrace.py --no-header /usr/share/qemu-kvm/trace-events /tmp/trace.log
-- 
1.8.3.1

